packages/expo-module-autolinking/src/index.ts

```diff
@@ -0,0 +1,103 @@
+import commander from 'commander';
+
+import {
+  findModulesAsync,
+  resolveModulesAsync,
+  verifySearchResults,
+  generatePackageListAsync,
+  mergeLinkingOptionsAsync,
+} from './autolinking';
+import { GenerateOptions, ResolveOptions, SearchOptions, SearchResults } from './types';
+
+/**
+ * Registers a command that only searches for available expo modules.
+ */
+function registerSearchCommand<OptionsType extends SearchOptions>(
+  commandName: string,
+  fn: (search: SearchResults, options: OptionsType) => any
+) {
+  return commander
+    .command(`${commandName} [paths...]`)
+    .option<string[] | null>(
+      '-i, --ignore-paths <ignorePaths...>',
+      'Paths to ignore when looking up for modules.',
+      (value, previous) => (previous ?? []).concat(value)
+    )
+    .option<string[] | null>(
+      '-e, --exclude <exclude...>',
+      'Package names to exclude when looking up for modules.',
+      (value, previous) => (previous ?? []).concat(value)
+    )
+    .action(async (searchPaths, providedOptions) => {
+      const options = await mergeLinkingOptionsAsync<OptionsType>({
+        ...providedOptions,
+        searchPaths,
+      });
+      const searchResults = await findModulesAsync(options);
+      return await fn(searchResults, options);
+    });
+}
+
+/**
+ * Registers a command that searches for modules and then resolves them for specific platform.
+ */
+function registerResolveCommand<OptionsType extends ResolveOptions>(
+  commandName: string,
+  fn: (search: SearchResults, options: OptionsType) => any
+) {
+  return registerSearchCommand<OptionsType>(commandName, fn).option(
+    '-p, --platform [platform]',
+    'The platform that the resulted modules must support. Available options: "ios", "android"',
+    'ios'
+  );
+}
+
+module.exports = async function(args: string[]) {
+  // Searches for available expo modules.
+  registerSearchCommand('search', async results => {
+    console.log(require('util').inspect(results, false, null, true));
+  });
+
+  // Checks whether there are no resolving issues in the current setup.
+  registerSearchCommand('verify', results => {
+    const numberOfDuplicates = verifySearchResults(results);
+    if (!numberOfDuplicates) {
+      console.log('✅ Everything is fine!');
+    }
+  });
+
+  // Searches for available expo modules and resolves the results for given platform.
+  registerResolveCommand('resolve', async (results, options) => {
```

Comment:
I'm not sure about `resolve` name here 🤔 
All this command is about it to print the detected state right? 🤔 
Why not to name it `check`/`doctor`? 🤔

Reply:
It doesn't check or "doctor" anything — its behavior can more or less be compared to how `node` and `npm` resolves from which path the module should be loaded 😉 It's one step forward to `search` command because it additionally resolves search results to the config for specific platform. Maybe `config` name would work here, but I wanted to get rid of using the same commands as rn-cli 😂